File: app/src/main/java/ru/magflayer/spectrum/data/android/ResourceManager.java
package ru.magflayer.spectrum.data.android;

import android.content.Context;
import android.content.res.Resources;
import android.support.annotation.PluralsRes;
import android.support.annotation.StringRes;

import java.io.IOException;
import java.io.InputStream;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class ResourceManager {

    private final Resources resources;

    @Inject
    public ResourceManager(final Context context) {
        this.resources = context.getResources();
    }

    public String getString(@StringRes final int id) {
        return resources.getString(id);
    }

    public String getQuantityString(@PluralsRes final int id, final int quantity) {
        return resources.getQuantityString(id, quantity);
    }

    public InputStream getAsset(final String assetName) throws IOException {
        return resources.getAssets().open(assetName);
    }
}

File: app/src/main/java/ru/magflayer/spectrum/data/android/ResourceManager.kt
package ru.magflayer.spectrum.data.android

import android.content.Context
import androidx.annotation.PluralsRes
import androidx.annotation.StringRes
import dagger.hilt.android.qualifiers.ApplicationContext
import java.io.IOException
import java.io.InputStream
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class ResourceManager @Inject constructor(
    @ApplicationContext context: Context,
) {

    private val resources = context.resources

    fun getString(@StringRes id: Int): String {
        return resources.getString(id)
    }

    fun getQuantityString(@PluralsRes id: Int, quantity: Int): String {
        return resources.getQuantityString(id, quantity)
    }

    @Throws(IOException::class)
    fun getAsset(assetName: String): InputStream {
        return resources.assets.open(assetName)
    }
}
